### PR TITLE
Allow multiple_steps to also return the coordinates.

### PR DESCRIPTION
**Pull request by proteneer**
Note that this changes the signature of multiple_steps return type to a tuple `(du_dls, xs)`

### CODE REVIEW

File: fe/estimator.py [hunk 1]
```diff
@@ -27,13 +27,49 @@ def unflatten(aux_data, children):
...
 
-def simulate(lamb, box, x0, v0, final_potentials, integrator, equil_steps, prod_steps, get_trajectory=False,
+def simulate(lamb, box, x0, v0, final_potentials, integrator, equil_steps, prod_steps,
     x_interval=1000, du_dl_interval=5):
     """
     Run a simulation and collect relevant statistics for this simulation.
 
     Parameters
     ----------
+    lamb: float
```
Comment (badisa) [inline fe/estimator.py:37]: 👍

File: timemachine/cpp/src/context.cu [hunk 2]
```diff
@@ -61,46 +63,66 @@ void Context::add_observable(Observable *obs) {
...
+    int store_x_freq) {
     unsigned long long *d_du_dl_buffer = nullptr;
     // try catch block is to deal with leaks in d_du_dl_buffer
-    if(store_du_dl_freq==0) {
-        store_du_dl_freq = lambda_schedule.size();
+    if(store_du_dl_freq <= 0) {
+        throw std::runtime_error("store_du_dl_freq <= 0");
+    }
+    if(store_x_freq <= 0) {
+        throw std::runtime_error("store_x_freq <= 0");
```
Comment (maxentile) [inline timemachine/cpp/src/context.cu:76]: Should we also catch here if the user requests an x buffer that would be too big? (say if len(lambda_schedule) = 1e7 and store_x_freq = 1)

**Review by maxentile — APPROVED** [review]
Looks great!

File: timemachine/cpp/src/context.cu (hunk 2 above)
Comment (proteneer) [reply timemachine/cpp/src/context.cu:76]: The limits are going to be GPU memory size dependent. These types of checks are best left on the application code side of things (there's lots of other ways to shoot one self in the foot). The default behavior here is not storing any frames, so I think we're pretty safe.

**Review by badisa — APPROVED** [review]

**Review by maxentile — APPROVED** [review]
Nice, +1 on including the `*_freq` -> `*_interval` change!

**Comment by proteneer** [comment]
Thanks for reviews, squash merging.

**Pull request merged**